server/routerlicious/packages/services-telemetry/src/lumberFormatters.ts

```diff
@@ -0,0 +1,40 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Lumberjack } from "./lumberjack";
+import { ILumberFormatter, LogLevel } from "./resources";
+import { Lumber } from "./lumber";
+
+export class SanitizationLumberFormatter implements ILumberFormatter {
+	private readonly sensitiveKeys = [
+		/cookie/i,
+		/passw(or)?d/i,
+		/^pw$/,
+		/^pass$/i,
+		/secret/i,
+		/token/i,
+		/api[._-]?key/i,
+		/session[._-]?id/i,
+	];
+
+	private readonly redactedStr = "[LUMBER_REDACTED]";
+
+	public transform(lumber: Lumber<string>): void {
+		if (lumber.logLevel === LogLevel.Error && lumber.exception) {
+			Object.keys(lumber.exception).forEach((keyStr, value) => {
+				if (typeof value === "object" && value !== null) {
+					this.transform(value);
```

Comment:
Is there any chance this might lead to an infinite recursive loop?

Reply:
Modified this to traverse the exception object. I believe the only way it can enter into an infinite loop is if the exception has a reference to itself inside. But I'm not sure how likely this is.